server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -332,6 +317,8 @@ private void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collecto
                 } catch (CollectionTerminatedException e) {
                     // collection was terminated prematurely
                     // continue with the following leaf
+                } catch (QueryPhase.TimeExceededException e) {
+                    searchContext.setSearchTimedOut(true);
```

Comment:
I don't think there's any more code that gets executed after this catch (and the one below), so I don't think `return` is necessary. However, I think it's probably still a good idea to add it in case additional logic is introduced later.